src/command.h

```diff
@@ -148,10 +148,23 @@ class PipelineCommand : public Command {
   Pipeline* pipeline_ = nullptr;
 };
 
+/// Base class for commands which contain a pipeline.
+class TimedPipelineCommand : public PipelineCommand {
+ public:
+  ~TimedPipelineCommand() override;
+
+  bool IsTimedExecution() const { return timed_execution_; }
+
+ protected:
+  TimedPipelineCommand(Type type, Pipeline* pipeline, bool timed_execution);
+
+  bool timed_execution_ = false;
+};
+
 /// Command to draw a rectangle on screen.
-class DrawRectCommand : public PipelineCommand {
+class DrawRectCommand : public TimedPipelineCommand {
  public:
-  DrawRectCommand(Pipeline* pipeline, PipelineData data);
+  DrawRectCommand(Pipeline* pipeline, PipelineData data, bool timed_execution);
```

Comment:
I think I'd prefer a `SetTimedExecution` similar to the setters for all the other params. Keep sit a bit more consistent with how the rest of the pipeline works. Can default to `false`, so we really only need to set it to enable it.

Reply:
Could also add the setter to the base class, which minimizes the number of changes in here.